doc/nrf/glossary.rst

```diff
@@ -0,0 +1,447 @@
+
+Signal-to-Noise Ratio (SNR)
+   The level of signal power compared to the level of noise power, often expressed in decibels (dB).
+
+Simple Network Time Protocol (SNTP)
+   A less complex implementation of NTP, using the same protocol but without requiring the storage of state over extended periods of time.
+
+Soft fork
+   A fork that contains a very small set of changes when compared to its upstream.
+
+SoftDevice
+   A wireless protocol stack that complements the nRF5 Series SoCs.
+   Nordic Semiconductor provides these stacks as qualified, precompiled binary files.
+
+SoftDevice Manager (SDM)
+   A SoftDevice component that controls the SoftDevice state and configures the behavior of certain core functionality.
+
+Software Development Kit (SDK)
+   A set of tools used for developing applications for a specific device or operating system.
+
+Spinel
+   A general management protocol for enabling a host device to communicate with and manage co-processors, like a network co-processor (NCP) or a radio co-processor (RCP).
+
+System in Package (SiP)
+   Several integrated circuits, often from different technologies, enclosed in a single module that performs as a system or subsystem.
+
+System on Chip (SoC)
+   A microchip that integrates all the necessary electronic circuits and components of a computer or other electronic systems on a single integrated circuit.
+
+System Protection Unit (SPU)
+   The central point in the system that controls access to memories, peripherals, and other resources.
+
+Target
+   The goal of an operation, for example, programming a specific image on a device, compiling a specific set of files, or removing previously generated files.
+
+Transmission Control Protocol (TCP)
+   A connection-oriented protocol that provides reliable transport.
+   This reliability comes at the cost of control packets overhead of the protocol itself, making it unsuitable for bandwidth-constrained applications.
+
+Toolchain
+   A set of development tools.
+   See `GNU Arm Embedded Toolchain`_.
+
+Transmit Data (TXD)
+   A signal line in a serial interface that transmits data to another device.
+
+Unicast addressing
+   An addressing type that uses a one-to-one association between the destination address and the network endpoint.
+   Each destination address uniquely identifies a single receiver endpoint.
+
+Universal Asynchronous Receiver/Transmitter (UART)
+   A hardware device for asynchronous serial communication between devices.
+
+Universal Serial Bus (USB)
+   An industry standard that establishes specifications for cables and connectors and protocols for connection, communication, and power supply between computers, peripheral devices, and other computers.
+
+Upmerge
+   The act of updating (synchronizing) a downstream repository with a newer revision of its upstream repository.
+
+Upstream branch
+   A branch located upstream, relative to another branch, in the flow of information.
```

Comment:
I would remove this altogether

Reply:
@carlescufi The idea to create separate entries for upstream branch and upstream repo has resulted from the fact that a single word _upstream_ might mean either repo or branch, depending on the context. Therefore it seemed natural to extend the documentation so that both meanings are covered. Why do you think it's better to remove it?

Reply:
Because usually we do not use branches in our repos. Instead, we use `master` or release branches, which can be translated to "releases".

Reply:
Same as above

Reply:
fixed